src/core/build/assets.ts

```diff
@@ -8,10 +8,17 @@ import { compressPublicAssets } from "../utils/compress";
 const NEGATION_RE = /^(!?)(.*)$/;
 const PARENT_DIR_GLOB_RE = /!?\.\.\//;
 
-export async function copyPublicAssets(nitro: Nitro) {
+export async function getPublicAssets(nitro: Nitro) {
   if (nitro.options.noPublicDir) {
     return;
   }
+
+  if (nitro._cachedPublicAssets) {
+    return nitro._cachedPublicAssets;
```

Comment:
I think this cache will be unreliable if nuxt emits assets after prerendering.

We can keep `copyPublicAssets` intact and introduce a util that scans only unprefixed public assets (usable for opt-in feature of pretender)